internal/common/tasks/scripts/push_artifact.sh

```
@@ -504,3 +519,31 @@ PYEOF
+  echo "Attaching $label ($(du -sh "$file" | cut -f1)) to ${repo_url}@${DISK_DIGEST}"
+  if ! "$HOME/bin/oras" attach \
+    --artifact-type "$artifact_type" \
+    "${repo_url}@${DISK_DIGEST}" \
+    "${file}:${artifact_type}"; then
+    echo "ERROR: Failed to attach $label (fatal in reproducible mode)"
+    exit 1
+  fi
+}
+
+if [ "$REPRODUCIBLE" = "true" ] && [ -n "$DISK_DIGEST" ]; then
+  cd /workspace/shared
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)